### ISSUE TITLE
Atom no longer restores previous windows on restart 

### ISSUE THREAD

**Issue by cdown512** [atom/atom#19343]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Up until a week or two ago, anytime I quit Atom, any tabs (including unsaved ones) I had open in the window were saved and re-opened the next time Atom started.  Now, when I quit Atom, if I have unsaved tabs, it prompts me to save the files, then after I make a decision on that and close the App, the next time it starts, it starts fresh with no retained files and an empty "untitled" page.  I have checked my Core settings and have the Restore Previous Windows On Start setting = 'always'.  I've tried it with 'yes' as well.  I've tried in safe mode as well.  The previous windows or tabs are never restored on a restart.  

### Steps to Reproduce

1. Have window with several tabs of files open in Atom, have "Restore Previous Windows On Start" setting = 'always'.
2. Quit Atom
3. Start Atom

**Expected behavior:** 
Atom does not prompt to save un-saved windows/tabs, re-opens with previous windows/tabs presented if Restore setting is set to 'yes' or 'always'

**Actual behavior:**
Atom wants to kill unsaved windows/tabs on shutdown by prompting to save, always re-opens with "untitled" file, and previous windows/tabs are not re-opened

**Reproduces how often:** 
100%

### Versions

1.37.0

### Additional Information

macOS 10.14.5

**Comment by rsese** [comment]
Thanks for the report!  I was unable to reproduce with 1.37.0 on macOS 10.12.6 - I tried with these steps (I have `Restore Previous Windows On Start` set to `always`):

1. `mkdir test; cd test`
2. `echo "hello world" > test.txt; echo "hello world" > text2.txt`
3. `atom --safe .`
4. Double click `test.txt` and `test2.txt` in the tree view to open them
5. Make an edit to each file (e.g. add a newline) so there's an unsaved change
6. Quit Atom `Cmd+q`

At this point Atom quits and I'm not prompted to save the files.  If I start Atom again with `atom --safe`, the test project is reopened with `test.txt` and `test2.txt` both open and with the unsaved changes from step 5.

Can you reproduce with these steps?  And if I misunderstood your report please let me know.

**Comment by cdown512** [comment]
Hey Robert.

Thanks so much for the reply.  Strangely, the behavior is no longer happening after following these steps.  I had to toggle on the Tree view in your steps (wasn’t displaying before), and then Open the “test” directory as a project to get the files to show up in the tree view.  But after I did that, I was able to open them and edit them, then quit Atom without getting prompted to save and re-open with the last state of the files.  So seems OK now.

Only things that have changed since yesterday were toggling the Tree view/loading the project folder in Atom, and the fact that I rebooted my laptop.

Thanks again for the help.

-Chris

On May 15, 2019, at 5:34 PM, Robert Sese <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for the report! I was unable to reproduce with 1.37.0 on macOS 10.12.6 - I tried with these steps (I have Restore Previous Windows On Start set to always):

  1.  mkdir test; cd test
  2.  echo "hello world" > test.txt; echo "hello world" > text2.txt
  3.  atom --safe .
  4.  Double click test.txt and test2.txt in the tree view to open them
  5.  Make an edit to each file (e.g. add a newline) so there's an unsaved change
  6.  Quit Atom Cmd+q

At this point Atom quits and I'm not prompted to save the files. If I start Atom again with atom --safe, the test project is reopened with test.txt and test2.txt both open and with the unsaved changes from step 5.

Can you reproduce with these steps? And if I misunderstood your report please let me know.

**Comment by rsese** [comment]
Huh that is odd, not sure what in particular made the behavior you were seeing before go away...

In any case, since you're not seeing the reported behavior anymore we'll close for now but if you  come across steps to reproduce or start seeing this again, we can always reopen.

**Comment by terencehill** [comment]
As a new user of Atom (Windows 10 here), I struggled for hours trying to understand why the heck the option "Restore previous window on start" wasn't working. I was about to thrash the whole application for this reason (I'm happy with jEdit anyway) when by chance I've found out that I needed to add a project folder in the tree view to make it work.

If there is no easy way to make it work without adding any project, can it at least be documented under "Restore previous window on start" that it works only if you add a project folder?
